altitude_parsing.c

```diff
@@ -0,0 +1,77 @@
+#include "altitude_parsing.h"
+
+static int32_t altitude = -999;
+static bool new_altitude = false;
+static char string[16];
+
+char rx_pool[32]; //32 bytes should be plenty
+
+static srb_ctx_t rx_buf;
+
+int32_t get_altitude(){
+    new_altitude = false;
+    return altitude;
+}
+
+bool new_altitude_available(){
+    return new_altitude;
+}
+
+void uart1_rx_init(uint32_t baud, uint32_t osc_freq){
+    // we will use -1 to indicate no recived altitude
+    altitude = -1;
```

Comment:
Nitpick/good practice: It's a bit odd that `altitude` is being initialized in this function, when the rest of the function deals exclusively with setting up the low level UART settings. It doesn't quite belong and can be decoupled.

The rest of the function looks nicely done at first glance.